spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/TtlScheduler.java

```diff
@@ -42,27 +43,36 @@
 
 	private final TaskScheduler scheduler = new ConcurrentTaskScheduler(Executors.newSingleThreadScheduledExecutor());
 
-	private HeartbeatProperties configuration;
+	private HeartbeatProperties heartbeatProperties;
+
+	private ConsulDiscoveryProperties discoveryProperties;
 
 	private ConsulClient client;
 
-	public TtlScheduler(HeartbeatProperties configuration, ConsulClient client) {
-		this.configuration = configuration;
+	private ReregistrationPredicate reregistrationPredicate;
+
+	private NewService registeredService;
+
+	public TtlScheduler(HeartbeatProperties heartbeatProperties, ConsulDiscoveryProperties discoveryProperties,
+			ConsulClient client, ReregistrationPredicate reregistrationPredicate) {
+		this.heartbeatProperties = heartbeatProperties;
+		this.discoveryProperties = discoveryProperties;
 		this.client = client;
+		this.reregistrationPredicate = reregistrationPredicate;
 	}
 
-	@Deprecated
 	public void add(final NewService service) {
 		add(service.getId());
+		this.registeredService = service;
```

Comment:
There's a reason add adds into a map. Any reason why `NewService` doesn't get added to one as well.

Reply:
Do you meath it should have `Map<String, NewService> registeredServices` ?

Reply:
Yes, I think so. You'll need to update remove as well.

Reply:
Updated